pkg/types/types.go

```diff
@@ -54,10 +54,5 @@ type SubmarinerSpecification struct {
 }
 
 func (subSpec *SubmarinerSpecification) GetIPFamilies() []k8snet.IPFamily {
-	ipFamilies := set.New[k8snet.IPFamily]()
-
-	// TODO_IPV6: set ipFamilies according to ClusterCidr content
-	ipFamilies.Insert(k8snet.IPv4)
-
-	return ipFamilies.UnsortedList()
+	return cidr.ExtractIPFamilies(subSpec.ClusterCidr)
```

Comment:
I think this change might break dual-stack CI gates , lets wait and see dual-stack tests results

Reply:
They all passed. They may start failing as more and more functionality is added, if so we can mark them as non-required.

Reply:
Hmm, for dual-stack gate, I don't understand how getPublicIP for IPv6 works ...
[1] 
https://github.com/submariner-io/submariner/blob/devel/pkg/endpoint/local_endpoint.go#L189

Reply:
`getPublicIP` returns an empty string and no error for IPv6: https://github.com/submariner-io/submariner/blob/959fb14c9b1d497f7cdb00b86046bc25a9b25fcf/pkg/endpoint/public_ip.go#L117. And `endpointSpec.SetPublicIP` ignores an empty string.